In [1]:
import pandas as pd

Загружаем таблицу с данными

In [10]:
df = pd.read_csv('DataScienceJobs.csv')

In [11]:
df # получение всей таблицы

name          address  \
0         Начинающий специалист по работе с нейросетям  Санкт-Петербург   
1                                 Junior Data Engineer  Нижний Новгород   
2                                           ML-инженер           Москва   
3                                       Data Scientist           Москва   
4                            Python Developer (Junior)            Томск   
..                                                 ...              ...   
355  Ведущий специалист по разработке скоринговых карт      Новосибирск   
356       Lead data scientist (Анализ цены и Trade in)           Москва   
357                              Senior Data Scientist            Томск   
358                            Data engineer (middle+)      Новосибирск   
359  Автор дисциплины "Задачи NLP в машинном обучении"  Нижний Новгород   

             experience                                             skills  \
0    От 1 года до 3 лет  Разработка технических заданий, Python, Сбор и...   
1    От 1 года до 3 лет                                   SQL, jvm, Hadoop   
2             Нет опыта  Знания математики, алгоритмов, структур данных...   
3    От 1 года до 3 лет  Python, Machine Learning, Математическое модел...   
4    От 1 года до 3 лет                            Python, Git, ООП, MySQL   
..                  ...                                                ...   
355  От 1 года до 3 лет  SQL, Sas, Базы данных, Скоринговая карта, База...   
356       От 3 до 6 лет  PyData: Pandas, NumPy, SciPy, scikit-learn, py...   
357         Более 6 лет  SQL, Анализ данных, A/B тесты, LTV prediction,...   
358       От 3 до 6 лет  Linux, ETL, Apache Kafka, Apache Spark, Apache...   
359       От 3 до 6 лет  NLP, ML, Data Science, Обучение и развитие, На...   

     sal_from  sal_to  
0       55000   60000  
1      150000  200000  
2      150000  170000  
3      180000  200000  
4       60000  110000  
..        ...     ...  
355     55000  105000  
356    180000  200000  
357    200000  210000  
358    150000  180000  
359    270000  320000  

[360 rows x 6 columns]

In [12]:
df.head() # получение первых пяти строк таблицы

name          address  \
0  Начинающий специалист по работе с нейросетям  Санкт-Петербург   
1                          Junior Data Engineer  Нижний Новгород   
2                                    ML-инженер           Москва   
3                                Data Scientist           Москва   
4                     Python Developer (Junior)            Томск   

           experience                                             skills  \
0  От 1 года до 3 лет  Разработка технических заданий, Python, Сбор и...   
1  От 1 года до 3 лет                                   SQL, jvm, Hadoop   
2           Нет опыта  Знания математики, алгоритмов, структур данных...   
3  От 1 года до 3 лет  Python, Machine Learning, Математическое модел...   
4  От 1 года до 3 лет                            Python, Git, ООП, MySQL   

   sal_from  sal_to  
0     55000   60000  
1    150000  200000  
2    150000  170000  
3    180000  200000  
4     60000  110000

In [13]:
df.tail() # получение последних пяти строк таблицы

name          address  \
355  Ведущий специалист по разработке скоринговых карт      Новосибирск   
356       Lead data scientist (Анализ цены и Trade in)           Москва   
357                              Senior Data Scientist            Томск   
358                            Data engineer (middle+)      Новосибирск   
359  Автор дисциплины "Задачи NLP в машинном обучении"  Нижний Новгород   

             experience                                             skills  \
355  От 1 года до 3 лет  SQL, Sas, Базы данных, Скоринговая карта, База...   
356       От 3 до 6 лет  PyData: Pandas, NumPy, SciPy, scikit-learn, py...   
357         Более 6 лет  SQL, Анализ данных, A/B тесты, LTV prediction,...   
358       От 3 до 6 лет  Linux, ETL, Apache Kafka, Apache Spark, Apache...   
359       От 3 до 6 лет  NLP, ML, Data Science, Обучение и развитие, На...   

     sal_from  sal_to  
355     55000  105000  
356    180000  200000  
357    200000  210000  
358    150000  180000  
359    270000  320000

In [14]:
df.sample(10) # получение случайных 10 строк из датафрейма

name          address  \
55                                      Data Scientist           Казань   
133                                     Data Scientist           Москва   
170             Data Scientist (Алгоритмы оптимизации)           Москва   
81                                      Data Scientist           Москва   
91                                Data Scientist (nlp)           Москва   
247                 Data Scientist (модели PD/LGD/EAD)  Нижний Новгород   
146                                      Data Engineer     Екатеринбург   
344                               DS NLP (LLM), Senior           Москва   
54   Cпециалист-исследователь в области обработки а...  Нижний Новгород   
89                        Middle/Senior Data Scientist           Москва   

             experience                                             skills  \
55        От 3 до 6 лет                                             Python   
133       От 3 до 6 лет  Уверенное владение Python. Опыт работы с библи...   
170       От 3 до 6 лет  Python, C++, Математическая статистика, Методы...   
81        От 3 до 6 лет  уверенное знание Python (библиотеки: NumPy, Pa...   
91        От 3 до 6 лет        ML, Python, CatBoost, STT, ASR, LabelStudio   
247  От 1 года до 3 лет            Python, ML, Data Science, Анализ данных   
146  От 1 года до 3 лет       Spark, Big Data, Hadoop, Базы данных, Python   
344       От 3 до 6 лет  Python, SQL, PySpark, Pandas, PyTorch, RecBole...   
54   От 1 года до 3 лет        Python, PyTorch, NLP, NER, Machine Learning   
89        От 3 до 6 лет                     Python, bash, git, SQL, Hadoop   

     sal_from  sal_to  
55      55000   85000  
133    280000  330000  
170    150000  220000  
81      90000  140000  
91     150000  180000  
247    100000  110000  
146    300000  350000  
344     85000  115000  
54     130000  160000  
89     180000  210000

Описание показателей: <br>name - название вакансии <br>address - город <br>experience - требуемый опыт <br>skills - требуемые навыки <br>sal_from - нижний порог зарплаты <br>sal_to - верхний порог зарплаты 

In [15]:
df.shape # кол-во строк и столбцов

(360, 6)

In [16]:
df.info() # сводная инофрмация о таблице

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 360 entries, 0 to 359
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   name        360 non-null    object
 1   address     360 non-null    object
 2   experience  360 non-null    object
 3   skills      360 non-null    object
 4   sal_from    360 non-null    int64 
 5   sal_to      360 non-null    int64 
dtypes: int64(2), object(4)
memory usage: 17.0+ KB


Таблица выше: 
1. `<class 'pandas.core.frame.DataFrame'>` – объект DataFrame
2. `RangeIndex: 360 entries, 0 to 359` – в таблице 360 строк с индексами от 0 до 359
3. `Data columns (total 6 columns):` – в таблице 6 колонок
4. После выводится небольшая таблица с информацией о названии колонки (`Column`), количестве непустых ячеек в колонке (`Non-Null Count`), а также тип данных (`Dtype`)
5. `dtypes: int64(2), object(4)` – сводка по количеству столбцов с разными типами данных
6. `memory usage: 17.0+ KB` – сколько памяти занимает таблица

Обращение к данным в таблице

`датафрейм['столбец']` или `датафрейм.столбец` – обращение к одному столбцу

`датафрейм[['столбец 1', 'столбец 2', ...]]` – обращение к нескольким столбцам

In [ ]:
df['name']
# или
df.name

0           Начинающий специалист по работе с нейросетям
1                                   Junior Data Engineer
2                                             ML-инженер
3                                         Data Scientist
4                              Python Developer (Junior)
                             ...                        
355    Ведущий специалист по разработке скоринговых карт
356         Lead data scientist (Анализ цены и Trade in)
357                                Senior Data Scientist
358                              Data engineer (middle+)
359    Автор дисциплины "Задачи NLP в машинном обучении"
Name: name, Length: 360, dtype: object

In [21]:
df[['name', 'address']]

name          address
0         Начинающий специалист по работе с нейросетям  Санкт-Петербург
1                                 Junior Data Engineer  Нижний Новгород
2                                           ML-инженер           Москва
3                                       Data Scientist           Москва
4                            Python Developer (Junior)            Томск
..                                                 ...              ...
355  Ведущий специалист по разработке скоринговых карт      Новосибирск
356       Lead data scientist (Анализ цены и Trade in)           Москва
357                              Senior Data Scientist            Томск
358                            Data engineer (middle+)      Новосибирск
359  Автор дисциплины "Задачи NLP в машинном обучении"  Нижний Новгород

[360 rows x 2 columns]

Обращение к строкам

`датафрейм.loc['название строки' или индекс]` – обращение к строке по названию \
`датафрейм.iloc[индекс]` – обращение к строке по индексу (считается с 0)

In [25]:
df.loc[0]

name               Начинающий специалист по работе с нейросетям
address                                         Санкт-Петербург
experience                                   От 1 года до 3 лет
skills        Разработка технических заданий, Python, Сбор и...
sal_from                                                  55000
sal_to                                                    60000
Name: 0, dtype: object

In [26]:
df.iloc[0]

name               Начинающий специалист по работе с нейросетям
address                                         Санкт-Петербург
experience                                   От 1 года до 3 лет
skills        Разработка технических заданий, Python, Сбор и...
sal_from                                                  55000
sal_to                                                    60000
Name: 0, dtype: object

Обращение к ячейкам

`датафрейм['столбец'][индекс строки]` или `датафрейм.loc[индекс]['столбец']`

In [27]:
df['name'][0]

'Начинающий специалист по работе с нейросетям'

In [28]:
df.loc[0]['name']

'Начинающий специалист по работе с нейросетям'

In [29]:
type(df['name'][0])

str

In [ ]:
df['sal_from'][0] # нижняя граница зарплаты в первой вакансии

np.int64(55000)

In [31]:
type(df['sal_from'][0])

numpy.int64